### ISSUE TITLE
Incorrect behavior when using certain keywords

### ISSUE THREAD

**Issue by garsuga** [cypress-io/cypress#22235]
### Current behavior

Write some code like
```
let x = {};
let self = { foo: x };
let parent = x;

console.log(parent == self.foo);
console.log(self.foo == parent);
console.log(parent === self.foo);
console.log(self.foo === parent);
```

Output is
```
false
true
false
true
```

Using code like
```
let x = {};
let self = { foo: x };
let Jparent = x;

console.log(Jparent == self.foo);
console.log(self.foo == Jparent);
console.log(Jparent === self.foo);
console.log(self.foo === Jparent);
```
yields this error on the first line
`ReferenceError: Jself is not defined`

This points towards the replacements done in [https://github.com/cypress-io/cypress/blob/56234e52d6d1cbd292acdfd5f5d547f0c4706b51/packages/proxy/lib/http/util/regex-rewriter.ts](https://github.com/cypress-io/cypress/blob/56234e52d6d1cbd292acdfd5f5d547f0c4706b51/packages/proxy/lib/http/util/regex-rewriter.ts)

Our code uses some variables named `self` and the issue above also occurs when using `someVar.parent` in place of `parent`

### Desired behavior

All 4 statements should print `true`

### Test code to reproduce

As expected any code written that matches the regular expressions defined in [https://github.com/cypress-io/cypress/blob/56234e52d6d1cbd292acdfd5f5d547f0c4706b51/packages/proxy/lib/http/util/regex-rewriter.ts](https://github.com/cypress-io/cypress/blob/56234e52d6d1cbd292acdfd5f5d547f0c4706b51/packages/proxy/lib/http/util/regex-rewriter.ts) will behave incorrectly. The replacements happen on code that does not refer to the values regex-rewriter needs to overwrite.

### Cypress Version

10.0.3

**Comment by marktnoonan** [comment]
Hi @garsuga, could you please provide a reproduction showing this issue? I tried running the code you provided in a test, and in my actual application, and in both cases all 4 statements logged as `true`. What is the context in which code needs to be running to hit this?

**Comment by mjhenkes** [comment]
@garsuga, have you had a chance to get a reproduction together?

**Comment by lmiller1990** [comment]
I'd be careful using `self` as a variable name - it's already defined globally on `window`, see [here](https://developer.mozilla.org/en-US/docs/Web/API/Window/self). 

I tried to reproduce by adding the code snippet to the test for our rewriter: https://github.com/cypress-io/cypress/blob/56234e52d6d1cbd292acdfd5f5d547f0c4706b51/packages/proxy/test/unit/http/util/regex-rewriter.spec.ts#L9 but no luck - it works as expected. Will wait on more information.

**Comment by mjhenkes** [comment]
Right now there doesn't seem to be enough information to reproduce the problem on our end. We'll have to close this issue until we can reproduce it. This does not mean that your issue is not happening - it just means that we do not have a path to move forward.

Please open a new issue with a reproducible example and link to this issue. Here are some tips for providing a [Short, Self Contained, Correct, Example](http://sscce.org/) and our own [Troubleshooting Cypress](https://on.cypress.io/troubleshooting) guide.
